craft_parts/plugins/rust_plugin.py

```diff
@@ -34,13 +35,16 @@
 
 logger = logging.getLogger(__name__)
 
-# A workaround for mypy false positives
-# see https://github.com/samuelcolvin/pydantic/issues/975#issuecomment-551147305
-# The proper fix requires Python 3.9+ (needs `typing.Annotated`)
-if TYPE_CHECKING:
-    UniqueStrList = list[str]
-else:
-    UniqueStrList = Annotated[List[str], Field(unique_items=True)]
+
+def _validate_list_is_unique(value: list):
```

Comment:
I was surprised to see only the rust plugin has a unique string list, especially since it makes the `after` keyword more specific than the `Part` model's `after` keyword.

Any reason not to proactively add this to `constraints.py`?

Reply:
That's coming in my next PR. I was trying to do minimal refactoring here.